backend/utils/natural_language_processing_api.py

```diff
@@ -40,5 +46,8 @@ def prompt_nlp_model(prompt):
     except APIConnectionError:
         return {"error": "No API KEY provided"}
 
-    # return the information into a json format
-    return json.loads(api_prompt.model_dump_json(indent=2))
+    # if organizing the response into FitScore doesn't work
+    except json.decoder.JSONDecodeError:
+        return {"error": "API Response was not in the correct format"}
```

Comment:
you should also catch 
```python
ValidationError
```
This is when the json_response can't be fit into FitScore model (due to improper format)